app/controllers/idv/personal_key_controller.rb

```diff
@@ -88,7 +88,11 @@ def profile
 
     def generate_personal_key
       cacher = Pii::Cacher.new(current_user, user_session)
-      profile.encrypt_recovery_pii(cacher.fetch)
+      new_personal_key = profile.encrypt_recovery_pii(cacher.fetch(profile.id))
+
+      profile.save
```

Comment:
Fairly certain this line resolves [LG-11743](https://cm-jira.usa.gov/browse/LG-11743)

Reply:
Because that's the only case where the profile isn't in the session? Otherwise, I'm wondering why that case is special.

Reply:
Let me double check, I'm pretty sure that the profile gets saved as a side effect of something else if the user does not log out before entering their GPO code

Reply:
Actually, a personal key is generated and stored in the session when the profile is minted. If you log out and log back in, that personal key is gone and _that_ is when `generate_personal_key` is called in `PersonalKeyController`. that currently generates the key but does not persist the profile with the re-encrypted data back to the db.